bigquery/tests/unit/test_client.py

```diff
@@ -106,11 +106,61 @@ def test_ctor_defaults(self):
 
         creds = _make_credentials()
         http = object()
-        client = self._make_one(project=self.PROJECT, credentials=creds, _http=http)
+        client = self._make_one(
+            project=self.PROJECT,
+            credentials=creds,
+            _http=http,
+            client_options={"api_endpoint": Connection.DEFAULT_API_ENDPOINT},
+        )
+        self.assertIsInstance(client._connection, Connection)
+        self.assertIs(client._connection.credentials, creds)
+        self.assertIs(client._connection.http, http)
+        self.assertIsNone(client.location)
+        self.assertEqual(
+            client._connection.API_BASE_URL, Connection.DEFAULT_API_ENDPOINT
+        )
+
+    def test_ctor_w_empty_client_options(self):
+        from google.cloud.bigquery._http import Connection
+        from google.api_core.client_options import ClientOptions
+
+        creds = _make_credentials()
+        http = object()
+        client_options = ClientOptions()
+        client = self._make_one(
+            project=self.PROJECT,
+            credentials=creds,
+            _http=http,
+            client_options=client_options,
+        )
+        self.assertIsInstance(client._connection, Connection)
```

Comment:
Since this test is for the `client_options` option, let's remove the asserts for the other Client properties.